spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -306,7 +306,7 @@ static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
+				return (originalContentType.startsWith(MimeTypeUtils.APPLICATION_JSON_VALUE)) ? MimeTypeUtils.APPLICATION_JSON
```

Comment:
FWIW we might always send back the originalContentType if it's not empty. The original change was meant to prevent the binder to override `application/json` but it seems like it shouldn't actually override anything.